### PR TITLE
chore(gatsby-core-utils): url to TS

### PR DESCRIPTION
**Pull request by danielkov**
Collection ticket for these tasks: #22041
☂ ticket: #21995

### CODE REVIEW

**Comment by blainekasten** [comment]
tests seem to be failing

**Comment by danielkov** [comment]
I think it's going to fail when the change to `index.js` is in a different PR.

File: packages/gatsby-core-utils/src/url.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-const path = require(`path`)
-const os = require(`os`)
+import path from "path"
+import os from "os"
 
 /**
  * @type {import('../index').urlResolve}
  */
-export function resolve(...segments) {
+export function resolve(...segments: string[]): string {
```
Comment (MichaelDeBoey) [inline packages/gatsby-core-utils/src/url.ts]:
```suggestion
export const resolve = (...segments: string[]): string => {
```

File: packages/gatsby-core-utils/src/url.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
-const path = require(`path`)
-const os = require(`os`)
+import path from "path"
+import os from "os"
 
 /**
  * @type {import('../index').urlResolve}
  */
```
Comment (MichaelDeBoey) [inline packages/gatsby-core-utils/src/url.ts]:
```suggestion
```

**Pull request merged**